demo/soma_clustering/src/soma_clustering.h

```diff
@@ -38,52 +38,64 @@ enum Substances { kSubstance0, kSubstance1 };
 inline int Simulate(int argc, const char** argv) {
   auto set_param = [](Param* param) {
     // Create an artificial bound for the simulation space
+    param->use_progress_bar = true;
     param->bound_space = Param::BoundSpaceMode::kClosed;
     param->min_bound = 0;
-    param->max_bound = 250;
-    param->unschedule_default_operations = {"mechanical forces"};
+    param->max_bound = 800;
+    // param->unschedule_default_operations = {"mechanical forces"};
   };
 
   Simulation simulation(argc, argv, set_param);
 
   // Define initial model
   auto* param = simulation.GetParam();
-  int num_cells = 20000;
+  int num_cells = 32000;
 
 #pragma omp parallel
   simulation.GetRandom()->SetSeed(4357);
 
   // Define the substances that cells may secrete
   // Order: substance id, substance_name, diffusion_coefficient, decay_constant,
   // and the resolution of the (finite difference) diffusion grid.
-  ModelInitializer::DefineSubstance(kSubstance0, "Substance_0", 50.0, 0.1, 20);
-  ModelInitializer::DefineSubstance(kSubstance1, "Substance_1", 50.0, 0.1, 20);
+  ModelInitializer::DefineSubstance(kSubstance0, "Substance_0", 0.05, 0.001,
+                                    40);
+  ModelInitializer::DefineSubstance(kSubstance1, "Substance_1", 0.05, 0.001,
+                                    40);
 
   int cell_type = 1;
   std::string substance_name = "Substance_0";
 
   auto construct = [&cell_type, &substance_name](const Real3& position) {
     auto* cell = new MyCell(position, cell_type);
     cell->SetDiameter(10);
-    cell->AddBehavior(new Secretion(substance_name));
-    cell->AddBehavior(new Chemotaxis(substance_name, 5));
+    cell->SetMass(0.1);
+    cell->AddBehavior(new Secretion(substance_name, 1));
+    cell->AddBehavior(new Chemotaxis(substance_name, 0.75));
     return cell;
   };
 
   // Construct num_cells/2 cells of type 0
-  ModelInitializer::CreateAgentsRandom(param->min_bound, param->max_bound,
+  ModelInitializer::CreateAgentsRandom(param->max_bound / 2 - 125,
+                                       param->max_bound / 2 + 125,
                                        num_cells / 2, construct);
   // Construct num_cells/2 cells of type 1
   cell_type = -1;
   substance_name = "Substance_1";
-  ModelInitializer::CreateAgentsRandom(param->min_bound, param->max_bound,
+  ModelInitializer::CreateAgentsRandom(param->max_bound / 2 - 125,
+                                       param->max_bound / 2 + 125,
                                        num_cells / 2, construct);
 
   // Run simulation for N timesteps
-  simulation.GetScheduler()->Simulate(1000);
+  const uint64_t timesteps = 1000;
+  if (timesteps < 6000) {
+    Log::Warning("SomaClustering",
+                 "We recommend to run the simulation for roughly 6000 time "
+                 "steps. Please change 'timesteps = ..'.");
+  }
+  simulation.GetScheduler()->Simulate(timesteps);
 
   // Check if criterion is met
```

Comment:
Let's rewrite the lines below as 
```
  // Check if criterion is met
  real_t spatial_range = 15;
  auto crit = GetCriterion(spatial_range, num_cells / 8);
  if (crit) {
    std::cout << "<SomaClustering> Clustering criterion met!" << std::endl;
  } else {
    std::cout << "<SomaClustering> Clustering criterion not met!" << std::endl;
  }
  std::cout << "Simulation completed successfully!\n";
  return 0;
```
Right now, the CI fails because the criterion is not met and returns `1` and does not say "simulation completed successfully". 

This should give us enough info if we ever have to debug this again, and users should understand that they need to set time_steps to `6000` to get better results.